girder/utility/webroot.py

```diff
@@ -57,6 +56,25 @@ def updateHtmlVars(self, vars):
         self.vars.update(vars)
         self.indexHtml = None
 
+    def addTemplateDirectory(self, templateDir):
+        """
+        Add a directory in which to search for templates.
+        """
+        self._templateLookup.directories.append(templateDir)
+
+    @property
+    def templateFilename(self):
+        """
+        The filename of the template to render. The file should be located in a
+        directory specified by addTemplateDirectory().
+        """
+        return self._templateFilename
+
+    @templateFilename.setter
+    def templateFilename(self, value):
+        self._templateFilename = value
+        self.indexHtml = None
```

Comment:
After https://github.com/girder/girder/pull/2274/commits/a5b6d83c10cbe91fcff0becb39f0e0f436589890 , it looks like `self.indexHtml` is a dead variable. I think you should remove it everywhere.